test/should_pass/any_pattern.erl

```diff
@@ -0,0 +1,14 @@
+-module(any_pattern).
+
+-export([pat_any/1]).
+
+%% This test case just increases coverage of add_any_types_pat
+%% with various patterns.
+%% We would expect that type of I, S and L is integer(), string() and
+%% list() respectively but currently they all become any()
+-spec pat_any(any()) -> ok.
+pat_any({$a}) -> ok;
+pat_any({[]}) -> ok;
+pat_any({<<I>>}) -> I;
+pat_any({S = "string"}) -> S;
+pat_any({"prefix" ++ L}) -> L.
```

Comment:
Add the cases with I, S, and L to a separate module in the `known_problems/should_fail` instead.

Reply:
I am happy to add them, but...
what do we consider as a known problem? definitely false positives (ie Gradualizer emits a type error although there is no type error). Also definitely when Gradualizer propagates the wrong type (that is not compatible with the real type)
But in this case Gradualizer is just not precise enough. There will always be error cases which Gradualizer cannot find, or when Gradualizer propagates a too general type. This I'd consider more like a `known_improvement_opportunity`, and has lower priority than fixing the known bugs.
(for the binary matching there is already a [TODO](https://github.com/josefs/Gradualizer/blob/master/src/typechecker.erl#L3686) in the code)

I don't try to bully here :) just your comment started me thinking. What do you guys think?

(maybe I will just add a `known_todo` dir that behaves exactly like `known_problems` and then problem solved :) )

Reply:
I think we can add it as a known problem. It is not as bad as a false positive, but we don't need to distinguish between more and less serious problems. Just give it a descriptive name.

The TODO for binaries can be known_problem test too if anyone cares to create one.